libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotMessageHandler.cs

```diff
@@ -17,9 +18,20 @@ protected override async Task<InvokeResponse> ProcessMessageRequestAsync(HttpReq
         {
             var activity = default(Activity);
 
-            using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8)))
+            // Get the request body and deserialize to the Activity object.
+            // We need to leave the stream open here so others downstream can access it.
+            var originalPosition = request.Body.Position;
+            request.Body.Position = 0;
+            try
             {
-                activity = BotMessageHandlerBase.BotMessageSerializer.Deserialize<Activity>(bodyReader);
+                using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8, true, 1024, true)))
```

Comment:
For these stray `bool`/`int` params that are not obvious what they are for unless you pull up intellisense (or have the entire BCL memorized 😄), please use named arguments so that it's very clear just glancing at the code.  Also... I wouldn't enable detecting encoding for this scenario, so just set that to `false`.


``` CSharp
new StreamReader(request.Body, Encoding.UTF8, detectEncodingFromByteOrderMarks: false, bufferSize: 1024, leaveOpen: true)
```